libraries/AP_Param/AP_Param.cpp

```diff
@@ -2329,6 +2340,36 @@ bool AP_Param::set_by_name(const char *name, float value)
     return false;
 }
 
+/*
+  get a value by name
+ */
+float AP_Param::get_by_name(const char *name)
```

Comment:
Return a bool, and do the float as a reference. This lets the script do nil punning, and allows you to distinguish between not finding the parameter, and a value of 0.

Reply:
Also any reason you can't rename this to just `get(const char *)` ?

Reply:
I have changed this to return bool, cant say I 100% understand how it works in lua but I copied how the battery amps function works and my test script works as expected. get_by_name makes scene in the context of ap param, would be nice to use simplified function names for lua, I can see how to change the lua name in the generated code, but no idea how to change the generator to allow a different name to be given in the .desc

Reply:
So whats happening here is the binding generator (because of the `Null` tag) is saying that any values that are annotated with it will only be valid if the function returns true. If it returns true then the function will push those values onto the lua stack so that the function can return them, if it's false then it will simply push a single nil value onto the stack. The main feature of this is it prevents you from forgetting to check the return type and using a random floating point value off the stack as if it was valid. (If you just blindly used the return type you'd eventually get an arithmetic error in most places, or a type error depending on what you did).

Reply:
Thank for the explanation